lib/src/formatters/iosStrings.ts

```diff
@@ -0,0 +1,142 @@
+import fetchText from "../http/textItems";
+import { ExportComponentsResponse, ExportTextItemsResponse, PullQueryParams, Variant } from "../http/types";
+import fetchComponents from "../http/components";
+import BaseFormatter from "./shared/base";
+import { applyMixins } from "./shared";
+import fetchProjects from "../http/projects";
+import IOSStringsOutputFile from "./shared/fileTypes/IOSStringsOutputFile";
+import fetchVariants from "../http/variants";
+
+interface ComponentsMap {
+  [variantId: string]: ExportComponentsResponse
+}
+interface TextItemsMap {
+  [projectId: string]: {
+    [variantId: string]: ExportTextItemsResponse
+  }
+}
+
+type IOSStringsAPIData = {
+  textItemsMap: TextItemsMap;
+  componentsMap: ComponentsMap;
+};
+
+export default class IOSStringsFormatter extends applyMixins(
+  BaseFormatter<IOSStringsOutputFile<{ variantId: string }>, IOSStringsAPIData>) {
+  private variants: { id: string }[] = [];
+
+  protected async fetchAPIData() {
+    await this.fetchVariants();
+    const textItemsMap = await this.fetchTextItemsMap();
+    const componentsMap = await this.fetchComponentsMap();
+
+    return { textItemsMap, componentsMap };
+  }
+
+  /**
+   * For each project/variant permutation and its fetched .strings data,
+   * create a new file with the expected naming
+   * 
+   * @returns {OutputFile[]} List of Output Files
+   */
+  protected transformAPIData(data: IOSStringsAPIData) {
```

Comment:
A little weird to call these "formatters" now that the export endpoint will do all of the formatting and all this is doing is saving to a file. Maybe worth renaming when you do the refactor work later.